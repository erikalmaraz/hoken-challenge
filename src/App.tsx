import React from 'react'
import Home from './pages/Home/Home'
import { Router, RouteComponentProps } from '@reach/router'

function App(): JSX.Element {
  return (
    <>
      <Router>
        <RouterPage path="/" pageComponent={<Home />} />
      </Router>
    </>
  )
}

const RouterPage = (
  props: { pageComponent: JSX.Element } & RouteComponentProps
): JSX.Element => props.pageComponent

export default App
