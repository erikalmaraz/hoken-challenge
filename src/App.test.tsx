import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import App from './App'

it('Render App component', () => {
  const component = render(<App />)
  expect(component).toMatchSnapshot()
})
