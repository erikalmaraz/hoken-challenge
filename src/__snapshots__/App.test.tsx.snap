// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="outline: none;"
        tabindex="-1"
      >
        <div
          class="sc-jSMfEi cCqStD"
        >
          <div
            class="sc-eCYdqJ iRyHKc"
          >
            <div
              class="sc-dkzDqf llzGsH"
            >
              <img
                alt="Equilibrium #3429 - By Jules Wyvern"
                src="./assets/images/equilibrium-bg.jpg"
              />
              <div
                class="image-card--layer"
              />
              <div
                class="image-card--more-info"
              >
                <img
                  alt="More information!"
                  src="./assets/images/icon-view.svg"
                />
              </div>
            </div>
            <div
              class="sc-hKMtZM jNerSk"
            >
              <div
                class="info-card--name"
              >
                <span>
                  Equilibrium #3429
                </span>
              </div>
              <div
                class="info-card--desc"
              >
                Our Equilibrium collection promotes balance and calm.
              </div>
            </div>
            <div
              class="sc-bczRLJ fUzKTP"
            >
              <div
                class="footer-card--amount"
              >
                <img
                  alt="Etherium icon"
                  src="./assets/images/icon-ethereum.svg"
                />
                <span
                  data-testid="footer-card--value"
                >
                  0.041
                   
                  ETH
                </span>
              </div>
              <div
                class="footer-card--time"
              >
                <img
                  alt="Clock icon"
                  src="./assets/images/icon-clock.svg"
                />
                <span
                  data-testid="footer-card--days"
                >
                  3
                   days left
                </span>
              </div>
            </div>
            <div
              class="sc-gsnTZi gFQbVV"
            >
              <div
                class="footer-card--avatar"
              >
                <img
                  alt="Jules Wyvern - Avatar"
                  src="./assets/images/image-avatar.png"
                />
              </div>
              <div
                class="footer-card--name"
              >
                Creation of 
                <b>
                  Jules Wyvern
                </b>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="outline: none;"
      tabindex="-1"
    >
      <div
        class="sc-jSMfEi cCqStD"
      >
        <div
          class="sc-eCYdqJ iRyHKc"
        >
          <div
            class="sc-dkzDqf llzGsH"
          >
            <img
              alt="Equilibrium #3429 - By Jules Wyvern"
              src="./assets/images/equilibrium-bg.jpg"
            />
            <div
              class="image-card--layer"
            />
            <div
              class="image-card--more-info"
            >
              <img
                alt="More information!"
                src="./assets/images/icon-view.svg"
              />
            </div>
          </div>
          <div
            class="sc-hKMtZM jNerSk"
          >
            <div
              class="info-card--name"
            >
              <span>
                Equilibrium #3429
              </span>
            </div>
            <div
              class="info-card--desc"
            >
              Our Equilibrium collection promotes balance and calm.
            </div>
          </div>
          <div
            class="sc-bczRLJ fUzKTP"
          >
            <div
              class="footer-card--amount"
            >
              <img
                alt="Etherium icon"
                src="./assets/images/icon-ethereum.svg"
              />
              <span
                data-testid="footer-card--value"
              >
                0.041
                 
                ETH
              </span>
            </div>
            <div
              class="footer-card--time"
            >
              <img
                alt="Clock icon"
                src="./assets/images/icon-clock.svg"
              />
              <span
                data-testid="footer-card--days"
              >
                3
                 days left
              </span>
            </div>
          </div>
          <div
            class="sc-gsnTZi gFQbVV"
          >
            <div
              class="footer-card--avatar"
            >
              <img
                alt="Jules Wyvern - Avatar"
                src="./assets/images/image-avatar.png"
              />
            </div>
            <div
              class="footer-card--name"
            >
              Creation of 
              <b>
                Jules Wyvern
              </b>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
